packages/block-editor/src/hooks/color-panel.js

```diff
@@ -12,12 +12,19 @@ import ContrastChecker from '../components/contrast-checker';
 import InspectorControls from '../components/inspector-controls';
 import { __unstableUseBlockRef as useBlockRef } from '../components/block-list/use-block-props/use-block-refs';
 
+const EMPTY_OBJECT = {};
+const DISABLE_GRADIENT_SETTINGS = {
```

Comment:
I'm a bit confused about the implementation of the color components, so I may be missing the whole context. I do wonder why we need to keep this data here? Don't we have access to `useSetting(color.customGradient)` and so on? I see that in use at `panel-color-gradient-settings.js`.

Reply:
We want to force the disable of gradients so we explicitly pass the settings that disable them. useSetting(color.customGradient) on paragraph may still return true if gradients are enabled but the gradients are not supported on the paragraph.

Reply:
I see. So what we're doing is passing those settings as empty to the underlying components when the block doesn't support gradients. Why don't we need to do the same for solids?

I also wonder if we could unwrap a bit all these layers/components and do the block support check earlier.

Reply:
> Why don't we need to do the same for solids?

Because we don't have a way of supporting gradients without supporting solid colors. Even solid colors are not supported the support mechanism is not triggered and the code never reaches here,.


> I also wonder if we could unwrap a bit all these layers/components and do the block support check earlier.

We do support checks early, for example, if colors are not supported our code never reaches here but to know if the component should not render at all we would need to inspect the palette of the specific block and other settings e.g: custom etc, etc and that is already done inside the PanelColorGradientSettings, we would be repeating all the checks this component does early which is a lot of code repetition. This solution seems to be the simplest one and the one that does not involve code repetition.